frame/timestamp/src/benchmarking.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Timestamp pallet benchmarking.
+
+use super::*;
+
+use sp_std::prelude::*;
+
+use frame_system::RawOrigin;
+use sp_runtime::{BenchmarkResults, BenchmarkParameter, selected_benchmark};
+use sp_runtime::traits::{Benchmarking, BenchmarkingSetup, Dispatchable};
+
+// Benchmark `set` extrinsic.
+struct Set;
+impl<T: Trait> BenchmarkingSetup<T, crate::Call<T>, RawOrigin<T::AccountId>> for Set {
+	fn components(&self) -> Vec<(BenchmarkParameter, u32, u32)> {
+		vec![]
+	}
+
+	fn instance(&self, _components: &[(BenchmarkParameter, u32)])
+		-> Result<(crate::Call<T>, RawOrigin<T::AccountId>), &'static str>
+	{
+		let user_origin = RawOrigin::None;
+		let now = 1.into();
+
+		// Return the `set` call
+		Ok((crate::Call::<T>::set(now), user_origin))
+	}
+}
+
+selected_benchmark!(Set);
+
+impl<T: Trait> Benchmarking<BenchmarkResults> for Module<T> {
+	fn run_benchmark(extrinsic: Vec<u8>, _steps: u32, repeat: u32) -> Result<Vec<BenchmarkResults>, &'static str> {
+		// Map the input to the selected benchmark.
+		let selected_benchmark = match extrinsic.as_slice() {
+			b"set" => SelectedBenchmark::Set,
+			_ => return Err("Could not find extrinsic."),
+		};
+		
+		// Warm up the DB
+		sp_io::benchmarking::commit_db();
+		sp_io::benchmarking::wipe_db();
+
+		// results go here
+		let mut results: Vec<BenchmarkResults> = Vec::new();
+		// Run the benchmark `repeat` times.
```

Comment:
```suggestion
		// Run the benchmark `repeat` times.
```

I am going to recommend some of these docs to be removed since I think over-documentation is also a problem. Probably thee come from Identity which was the first one and it was a bit overly-documented..

Reply:
Dumb question, whats wrong with "over" docmentation?

Reply:
Reduced readability. I also don't recall the fancy names of software engineering literature nowadays, but there is a thing with the context of _a good could should be documented via the code itself_. i.e. it shouldn't need that much documentation. 

Either way, this might be a bit of a personal taste, but IMO strongly: documentation is not like _the more the better_. 

Yes, no documentation is also bad. I am not denying this.

Reply:
i.e. 

```
// Wipe the DB back to the genesis state.
sp_io::benchmarking::wipe_db();
```

the name of the function here is really enough. Even if not, the reader should go to the definition of the function to read it. Explaining what a --for example-- function does at __each and every call site__ is too much.

Reply:
(I hope I have convinced you enough :D)

Reply:
I disagree at least with the very specific thing you pointed out.

It is not clear that `wipe_db` brings the runtime state back to genesis versus _actually_ wiping the DB clean.

Usually I let to tell a story with my code, and I like to make sure that story is told in words too. Sometimes the words mean one thing but the code does another. Writing both is an easy way to allow others to catch errors.

Reply:
Code should be as much as possible self explanational. If you need to document each line, there is something wrong with your code. (there are exceptions, like when you do really hard shit ^^)

However, saying that you iterate x times is obvious and is just unnecessary.

For the `wipe_db` function, the name is probably already chosen incorrectly. If it resets to genesis, it should not be called `wipe`. It could be `reset_db` (tells you that we reset to the old state) or even more expressive `reset_db_to_genesis` or just `reset_to_genesis`. This makes it directly clear what happens. For `wipe_db` you now always need to repeat yourself.

Reply:
EOD, there's a bit of personal preference into it. I am more leaned toward minimising inline doc. Fair enough and thanks to both for the good discussion 👍